python/pyspark/sql/streaming/readwriter.py

```diff
@@ -1122,6 +1122,85 @@ def trigger(
         self._jwrite = self._jwrite.trigger(jTrigger)
         return self
 
+    @staticmethod
+    def construct_foreach_function(f: Union[Callable[[Row], None], "SupportsProcess"]):
```

Comment:
Nice, thanks for reusing it.
The name should start with `_` since it is meant for internal use.